File: 66.Plus One.py
class Solution(object):
    def plusOne(self, digits):
        """
        :type digits: List[int]
        :rtype: List[int]
        """
        i=len(digits)-1
        while i>=0:
            if digits[i]!=9:
                break
            digits[i]=0
            i-=1
        if i==-1:
            digits.insert(0,1)
        else:
            digits[i]+=1
        return digits
        

File: 66.Plus One.rb
# @param {Integer[]} digits
# @return {Integer[]}
def plus_one(digits)
    carry = 1
    for i in (digits.size - 1).downto(0)
        digits[i] += carry
        carry = digits[i] / 10
        digits[i] %= 10
    end
    digits.insert(0, 1) if carry == 1
    return digits
end
